grc: Terminate the process instead of Killing with Terminate button (F7)

kartikp1995 opened the pull request:
Currently, GRC terminate button (F7) sends SIGKILL to the ongoing Python program.

This PR is an attempt to send SIGTERM signal in order to allow cleanup before termination. Also, sends SIGKILL signal to the process after 4 seconds if Python program couldn't terminate with SIGTERM.

Since, many hardware takes around 2 seconds to disconnect, 4 seconds should be more than enough for cleaning up.

jmcorgan commented:
Can you confirm the gobject import does add any new package dependencies?

kartikp1995 commented:
@jmcorgan It does not any extra dependency. The `gobject` is already being used in `grc/gui/Executor.py` file.

skoslowski commented on grc/gui/NotebookPage.py:
```diff
@@ -152,6 +153,30 @@ def set_proc(self, process):
+                False to stop timeout_add periodic calls
+            """
+            is_terminated = process.poll()
+            if is_terminated == None:
```
Always use ```is None```

kartikp1995 commented:
@skoslowski 4 seconds was just my guess about general shutdown time of the hardware. I have updated it to 2 seconds. I hope it's not too short!

Also, updated `== None` with `is None`. Thanks for the suggestion.

primercuervo commented on grc/gui/NotebookPage.py:
```diff
@@ -152,6 +153,30 @@ def set_proc(self, process):
+        Terminate the subprocess object
+
+        Add a callback to kill the process
+        after 4 seconds if not already terminated
```
Might not be at all important, but as you changed the time in the code, you might as well change it in the docstring.

kartikp1995 replied:
Oh silly me! Thanks! :+1:
